From 20c646364fd7703f8c54a54b42e9a7659b361023 Mon Sep 17 00:00:00 2001
From: Luca Palmieri <20745048+LukeMathWalker@users.noreply.github.com>
Date: Wed, 23 Jul 2025 13:02:22 +0200
Subject: [PATCH 05/34] Route import annotation

---
 app/src/blueprint.rs | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/app/src/blueprint.rs b/app/src/blueprint.rs
index 5271657..ba6d0e9 100644
--- a/app/src/blueprint.rs
+++ b/app/src/blueprint.rs
@@ -1,8 +1,10 @@
 use crate::telemetry;
+// px:route_import:start
 use pavex::{Blueprint, blueprint::from};
 
 /// The main blueprint, defining all the components used in this API.
 pub fn blueprint() -> Blueprint {
+    // px::skip:start
     let mut bp = Blueprint::new();
     // Bring into scope constructors, error handlers, configuration
     // and prebuilt types defined in the following crates
@@ -16,6 +18,8 @@ pub fn blueprint() -> Blueprint {
 
     telemetry::instrument(&mut bp);
 
+    // px::skip:end
     bp.prefix("/api").routes(from![crate]);
-    bp
+    bp // px::skip
 }
+// px:route_import:end
-- 
2.39.5 (Apple Git-154)

