From a749c190ccd615620d4ebac2eb21f45ad9458fff Mon Sep 17 00:00:00 2001
From: Luca Palmieri <20745048+LukeMathWalker@users.noreply.github.com>
Date: Wed, 23 Jul 2025 13:06:21 +0200
Subject: [PATCH 10/32] Extract path parameters in greet

---
 app/src/routes/greet.rs | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/app/src/routes/greet.rs b/app/src/routes/greet.rs
index 9a82e41..2e90f24 100644
--- a/app/src/routes/greet.rs
+++ b/app/src/routes/greet.rs
@@ -1,6 +1,12 @@
-use pavex::{get, response::Response};
+use pavex::{get, request::path::PathParams, response::Response};
+
+#[PathParams]
+pub struct GreetParams {
+    pub name: String,
+}
 
 #[get(path = "/greet/{name}")]
-pub fn greet() -> Response {
-    todo!()
+pub fn greet(params: PathParams<GreetParams>) -> Response {
+    let GreetParams { name } = params.0;
+    Response::ok().set_typed_body(format!("Hello, {name}!"))
 }
-- 
2.39.5 (Apple Git-154)

