From 9b3b116b450eac21b7fe3701398812c3b22cb418 Mon Sep 17 00:00:00 2001
From: Luca Palmieri <20745048+LukeMathWalker@users.noreply.github.com>
Date: Wed, 23 Jul 2025 15:48:49 +0200
Subject: [PATCH 19/34] Fallible user agent constructor

---
 app/src/user_agent.rs | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/app/src/user_agent.rs b/app/src/user_agent.rs
index 2ed7734..fd418a4 100644
--- a/app/src/user_agent.rs
+++ b/app/src/user_agent.rs
@@ -1,4 +1,4 @@
-use pavex::http::header::USER_AGENT;
+use pavex::http::header::{ToStrError, USER_AGENT};
 use pavex::methods;
 use pavex::request::RequestHead;
 
@@ -12,14 +12,11 @@ pub enum UserAgent {
 #[methods]
 impl UserAgent {
     #[request_scoped]
-    pub fn extract(request_head: &RequestHead) -> Self {
+    pub fn extract(request_head: &RequestHead) -> Result<Self, ToStrError> {
         let Some(user_agent) = request_head.headers.get(USER_AGENT) else {
-            return Self::Unknown;
+            return Ok(Self::Unknown);
         };
 
-        match user_agent.to_str() {
-            Ok(s) => Self::Known(s.into()),
-            Err(_e) => todo!(),
-        }
+        user_agent.to_str().map(|s| UserAgent::Known(s.into()))
     }
 }
-- 
2.39.5 (Apple Git-154)

