From fec513fb014b830cc23646af2ddeda0121e77255 Mon Sep 17 00:00:00 2001
From: Luca Palmieri <20745048+LukeMathWalker@users.noreply.github.com>
Date: Wed, 23 Jul 2025 15:53:25 +0200
Subject: [PATCH 24/34] Error handler for invalid user agent

---
 app/src/user_agent.rs | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/app/src/user_agent.rs b/app/src/user_agent.rs
index fd418a4..dcb954d 100644
--- a/app/src/user_agent.rs
+++ b/app/src/user_agent.rs
@@ -1,6 +1,7 @@
 use pavex::http::header::{ToStrError, USER_AGENT};
-use pavex::methods;
 use pavex::request::RequestHead;
+use pavex::response::Response;
+use pavex::{error_handler, methods};
 
 pub enum UserAgent {
     /// No `User-Agent` header was provided.
@@ -20,3 +21,9 @@ impl UserAgent {
         user_agent.to_str().map(|s| UserAgent::Known(s.into()))
     }
 }
+
+#[error_handler]
+pub fn invalid_user_agent(_e: &ToStrError) -> Response {
+    let body = "The `User-Agent` header value can only use ASCII printable characters.";
+    Response::bad_request().set_typed_body(body)
+}
-- 
2.39.5 (Apple Git-154)

