From 7f925c4ee53aa0749fc32e12a0e9658c4e9b36b6 Mon Sep 17 00:00:00 2001
From: Luca Palmieri <20745048+LukeMathWalker@users.noreply.github.com>
Date: Wed, 23 Jul 2025 15:57:23 +0200
Subject: [PATCH 33/34] Greet test annotations

---
 server/tests/integration/greet.rs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/server/tests/integration/greet.rs b/server/tests/integration/greet.rs
index 5a5d7f5..947ca6f 100644
--- a/server/tests/integration/greet.rs
+++ b/server/tests/integration/greet.rs
@@ -19,6 +19,7 @@ async fn greet_happy_path() {
     assert_eq!(response.text().await.unwrap(), "Hello, Ursula!");
 }
 
+// px:greet_unhappy_test:start
 #[tokio::test]
 async fn non_utf8_user_agent_is_rejected() {
     let api = TestApi::spawn().await;
@@ -38,3 +39,4 @@ async fn non_utf8_user_agent_is_rejected() {
         "The `User-Agent` header value can only use ASCII printable characters."
     );
 }
+// px:greet_unhappy_test:end
-- 
2.39.5 (Apple Git-154)

