diff --git a/demo/src/blueprint.rs b/demo/src/blueprint.rs
--- a/demo/src/blueprint.rs
+++ b/demo/src/blueprint.rs
@@ -4,6 +4,7 @@ use pavex::f;
 
 /// The main blueprint, containing all the routes, constructors and error handlers
 /// required by our API.
+// --8<-- [start:blueprint_definition]
 pub fn blueprint() -> Blueprint {
     let mut bp = Blueprint::new();
     register_common_constructors(&mut bp);
@@ -13,6 +14,7 @@ pub fn blueprint() -> Blueprint {
     bp.route(GET, "/api/ping", f!(crate::routes::status::ping));
     bp
 }
+// --8<-- [end:blueprint_definition]
 
 /// Common constructors used by all routes.
 fn register_common_constructors(bp: &mut Blueprint) {
