diff --git a/demo/src/routes/greet.rs b/demo/src/routes/greet.rs
index 38ec1e3..adfbbd5 100644
--- a/demo/src/routes/greet.rs
+++ b/demo/src/routes/greet.rs
@@ -1,5 +1,11 @@
+use pavex::request::RouteParams;
 use pavex::response::Response;
 
-pub fn greet() -> Response {
+#[RouteParams]
+pub struct GreetParams {
+    pub name: String,
+}
+
+pub fn greet(params: RouteParams<GreetParams> /* (2)! */) -> Response {
     todo!()
 }
