diff --git a/app/src/routes/greet.rs b/app/src/routes/greet.rs
index e0be313..5080f53 100644
--- a/app/src/routes/greet.rs
+++ b/app/src/routes/greet.rs
@@ -3,9 +3,11 @@ use pavex::response::Response;

 #[PathParams]
 pub struct GreetParams {
-    pub name: String, /* (1)! */
+    pub name: String,
 }

-pub fn greet(params: PathParams<GreetParams> /* (2)! */) -> Response {
-    todo!()
+pub fn greet(params: PathParams<GreetParams>) -> Response {
+    let GreetParams { name }/* (1)! */ = params.0;
+    Response::ok() // (2)!
+        .set_typed_body(format!("Hello, {name}!")) // (3)!
 }
