diff --git a/demo/src/blueprint.rs b/demo/src/blueprint.rs
--- a/demo/src/blueprint.rs
+++ b/demo/src/blueprint.rs
@@ -11,11 +11,7 @@ pub fn blueprint() -> Blueprint {
     add_telemetry_middleware(&mut bp);
 
     bp.route(GET, "/api/ping", f!(crate::routes::status::ping));
-    bp.route(
-        GET,
-        "/api/greet/:name", /* (1)! */
-        f!(crate::routes::greet::greet),
-    );
+    bp.route(GET, "/api/greet/:name", f!(crate::routes::greet::greet));
     bp
 }
 
diff --git a/demo/src/routes/greet.rs b/demo/src/routes/greet.rs
--- a/demo/src/routes/greet.rs
+++ b/demo/src/routes/greet.rs
@@ -8,7 +8,7 @@ pub struct GreetParams {
     pub name: String,
 }
 
-pub fn greet(params: RouteParams<GreetParams>, user_agent: UserAgent /* (1)! */) -> Response {
+pub fn greet(params: RouteParams<GreetParams>, user_agent: UserAgent) -> Response {
     if let UserAgent::Unknown = user_agent {
         return Response::unauthorized()
             .set_typed_body("You must provide a `User-Agent` header");
