diff --git a/app/src/user_agent.rs b/app/src/user_agent.rs
--- a/app/src/user_agent.rs
+++ b/app/src/user_agent.rs
@@ -1,6 +1,25 @@
+use pavex::http::header::USER_AGENT;
+use pavex::methods;
+use pavex::request::RequestHead;
+
 pub enum UserAgent {
     /// No `User-Agent` header was provided.
     Unknown,
     /// The value of the `User-Agent` header for the incoming request.
     Known(String),
 }
+
+#[methods]
+impl UserAgent {
+    #[request_scoped]
+    pub fn extract(request_head: &RequestHead) -> Self {
+        let Some(user_agent) = request_head.headers.get(USER_AGENT) else {
+            return Self::Unknown;
+        };
+
+        match user_agent.to_str() {
+            Ok(s) => Self::Known(s.into()),
+            Err(_e) => todo!(),
+        }
+    }
+}
