diff --git a/demo/src/user_agent.rs b/demo/src/user_agent.rs
index c9d0771..bb1f65b 100644
--- a/demo/src/user_agent.rs
+++ b/demo/src/user_agent.rs
@@ -1,4 +1,4 @@
-use pavex::http::header::USER_AGENT;
+use pavex::http::header::{ToStrError, USER_AGENT};
 use pavex::request::RequestHead;
 
 pub enum UserAgent {
@@ -9,14 +9,11 @@ pub enum UserAgent {
 }
 
 impl UserAgent {
-    pub fn extract(request_head: &RequestHead) -> Self {
+    pub fn extract(request_head: &RequestHead) -> Result<Self, ToStrError /* (1)! */> {
         let Some(user_agent) = request_head.headers.get(USER_AGENT) else {
-            return Self::Unknown;
+            return Ok(UserAgent::Unknown);
         };
 
-        match user_agent.to_str() {
-            Ok(s) => Self::Known(s.into()),
-            Err(_e) => todo!(),
-        }
+        user_agent.to_str().map(|s| UserAgent::Known(s.into()))
     }
 }
