diff --git a/demo/src/blueprint.rs b/demo/src/blueprint.rs
index e3623cd..c431f69 100644
--- a/demo/src/blueprint.rs
+++ b/demo/src/blueprint.rs
@@ -11,7 +11,8 @@ pub fn blueprint() -> Blueprint {
     bp.constructor(
         f!(crate::user_agent::UserAgent::extract),
         Lifecycle::RequestScoped,
-    );
+    )
+        .error_handler(f!(crate::user_agent::invalid_user_agent));
 
     add_telemetry_middleware(&mut bp);
 
diff --git a/demo/src/user_agent.rs b/demo/src/user_agent.rs
index bb1f65b..b1c935c 100644
--- a/demo/src/user_agent.rs
+++ b/demo/src/user_agent.rs
@@ -17,3 +17,9 @@ impl UserAgent {
         user_agent.to_str().map(|s| UserAgent::Known(s.into()))
     }
 }
+
+pub fn invalid_user_agent(_e: &ToStrError) -> Response {
+    Response::bad_request()
+        .set_typed_body("The `User-Agent` header value must be a valid UTF-8 string")
+        .box_body()
+}
