diff --git a/src/user.rs b/src/user.rs
index c40a4e9..a896a49 100644
--- a/src/user.rs
+++ b/src/user.rs
@@ -1,3 +1,5 @@
+use pavex::request::RequestHead;
+
 pub enum User {
     Anonymous,
     Authenticated(AuthenticatedUser),
@@ -8,7 +10,7 @@ pub struct AuthenticatedUser {
 }
 
 impl User {
-    pub fn extract() -> Self {
+    pub fn extract(request_head: &RequestHead) -> Self {
         todo!() // Business logic goes here
     }
 }
\ No newline at end of file
