diff --git a/src/blueprint.rs b/src/blueprint.rs
index 051b47b..9aad912 100644
--- a/src/blueprint.rs
+++ b/src/blueprint.rs
@@ -4,7 +4,7 @@ use pavex::f;
 
 pub fn blueprint() -> Blueprint {
     let mut bp = Blueprint::new();
-    bp.constructor(f!(crate::User::extract), Lifecycle::RequestScoped);
+    bp.request_scoped(f!(crate::User::extract));
     bp.wrap(f!(crate::authentication::reject_anonymous));
     bp.route(GET, "/greet", f!(crate::routes::greet));
     bp
