diff --git a/app/src/routes/greet.rs b/app/src/routes/greet.rs
index 38ec1e3..adfbbd5 100644
--- a/app/src/routes/greet.rs
+++ b/app/src/routes/greet.rs
@@ -1,5 +1,11 @@
+use pavex::request::path::PathParams;
 use pavex::response::Response;
 
-pub fn get() -> Response {
+#[PathParams]
+pub struct GreetParams {
+    pub name: String /* (1)! */
+}
+
+pub fn get(params: PathParams<GreetParams> /* (2)! */) -> Response {
     todo!()
 }
